iceoryx2-bb/lock-free/src/spmc/unrestricted_atomic.rs

```diff
@@ -51,6 +51,17 @@ impl<T: Copy> Producer<'_, T> {
     pub fn store(&self, new_value: T) {
         self.atomic.store(new_value);
     }
+
+    #[doc(hidden)]
+    pub unsafe fn get_ptr_to_write_cell(&self) -> *mut T {
```

Comment:
Even when the method is hidden, could you please add a `// # Safety` section here and define all contracts that the user must ensure that they are fulfilled.
I assume that this memory position shall not be modified after `update_write_cell` was called.
Maybe there is even more.

The same goes for `update_write_cell`. Could you maybe even add the prefix `__internal_` to those two methods to make it really clear that those shall not be used.
When reviewing some piece of code one does not see directly the `#[doc(hidden)]` annotation, just the pure function call and then an `__internal_` prefix becomes another signaling flag.